File: modifier_professeur.php
<!DOCTYPE html>
<html>
<head>
    <title>Modifier un professeur</title>
</head>
<body>
    <h1>Modifier un professeur</h1>

    <?php
    // Vérifier si l'ID du professeur est passé en paramètre
    if (isset($_GET['id'])) {
        $id_professeur = $_GET['id'];

        // Connexion à la base de données
        $dsn = "mysql:host=localhost;dbname=gdn";
        $username = "root";
        $password = "";

        try {
            $db = new PDO($dsn, $username, $password);
            $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);

            // Récupérer les détails du professeur
            $query = "SELECT * FROM professeurs WHERE ID_Professeur = :id";
            $stmt = $db->prepare($query);
            $stmt->bindValue(':id', $id_professeur);
            $stmt->execute();
            $professeur = $stmt->fetch(PDO::FETCH_ASSOC);

            if (!$professeur) {
                die("Professeur introuvable.");
            }

            // Traitement du formulaire de modification
            if ($_SERVER['REQUEST_METHOD'] === 'POST') {
                // Récupérer les nouvelles valeurs du formulaire
                $nom = $_POST['nom'];
                $prenom = $_POST['prenom'];
                $situation_matrimoniale = $_POST['situation_matrimoniale'];
                $lieu_naissance = $_POST['lieu_naissance'];
                $date_naissance = $_POST['date_naissance'];
                $diplome_formation = $_POST['diplome_formation'];
                $autre_diplome = $_POST['autre_diplome'];
                $matieres_enseignees = $_POST['matieres_enseignees'];
                $autre_matiere = $_POST['autre_matiere'];
                $login = $_POST['login'];
                $passwd = $_POST['passwd'];

                // Mettre à jour les informations du professeur dans la base de données
                $query = "UPDATE professeurs SET Nom_Professeur = :nom, Prenom_Professeur = :prenom, Situation_Matrimoniale = :situation_matrimoniale, Lieu_Naissance = :lieu_naissance, Date_Naissance = :date_naissance, Diplome_Formation = :diplome_formation, Autre_Diplome = :autre_diplome, Matieres_Enseignees = :matieres_enseignees, Autre_Matiere = :autre_matiere, Login_Professeur = :login, Passwd_Professeur = :passwd WHERE ID_Professeur = :id";
                $stmt = $db->prepare($query);
                $stmt->bindValue(':nom', $nom);
                $stmt->bindValue(':prenom', $prenom);
                $stmt->bindValue(':situation_matrimoniale', $situation_matrimoniale);
                $stmt->bindValue(':lieu_naissance', $lieu_naissance);
                $stmt->bindValue(':date_naissance', $date_naissance);
                $stmt->bindValue(':diplome_formation', $diplome_formation);
                $stmt->bindValue(':autre_diplome', $autre_diplome);
                $stmt->bindValue(':matieres_enseignees', $matieres_enseignees);
                $stmt->bindValue(':autre_matiere', $autre_matiere);
                $stmt->bindValue(':login', $login);
                $stmt->bindValue(':passwd', $passwd);
                $stmt->bindValue(':id', $id_professeur);

                $stmt->execute();

                echo "Les informations du professeur ont été mises à jour avec succès.";
            }
        } catch (PDOException $e) {
            die("Erreur : " . $e->getMessage());
        }
    } else {
        die("ID du professeur non spécifié.");
    }
    ?>

    <!-- Formulaire de modification -->
    <form method="POST" action="">
        <label for="nom">Nom:</label>
        <input type="text" id="nom" name="nom" value="<?php echo $professeur['Nom_Professeur']; ?>"><br>

        <label for="prenom">Prénom:</label>
        <input type="text" id="prenom" name="prenom" value="<?php echo $professeur['Prenom_Professeur']; ?>"><br>

        <label for="situation_matrimoniale">Situation Matrimoniale :</label>
        <input type="text" id="situation_matrimoniale" name="situation_matrimoniale" list="marital_status" required><br>
        <datalist id="marital_status">
        <option value="Célibataire">
        <option value="Marié(e)">
        <option value="Veuf(ve)">
        </datalist>

        <label for="lieu_naissance">Lieu de Naissance:</label>
        <input type="text" id="lieu_naissance" name="lieu_naissance" value="<?php echo $professeur['Lieu_Naissance']; ?>"><br>

        <label for="date_naissance">Date de Naissance:</label>
        <input type="text" id="date_naissance" name="date_naissance" value="<?php echo $professeur['Date_Naissance']; ?>"><br>

        <label for="diplome_formation">Diplôme de Formation:</label>
        <input type="text" id="diplome_formation" name="diplome_formation" value="<?php echo $professeur['Diplome_Formation']; ?>"><br>

        <label for="autre_diplome">Autre Diplôme:</label>
        <input type="text" id="autre_diplome" name="autre_diplome" value="<?php echo $professeur['Autre_Diplome']; ?>"><br>

        <label for="matieres_enseignees">Matières Enseignées:</label>
        <input type="text" id="matieres_enseignees" name="matieres_enseignees" value="<?php echo $professeur['Matieres_Enseignees']; ?>"><br>

        <label for="autre_matiere">Autre Matière:</label>
        <input type="text" id="autre_matiere" name="autre_matiere" value="<?php echo $professeur['Autre_Matiere']; ?>"><br>

        <label for="login">Login:</label>
        <input type="text" id="login" name="login" value="<?php echo $professeur['Login_Professeur']; ?>"><br>

        <label for="passwd">Mot de passe:</label>
        <input type="password" id="passwd" name="passwd" value="<?php echo $professeur['Passwd_Professeur']; ?>"><br>

        <input type="submit" value="Enregistrer">
    </form>

    <a href="gestion_professeurs.php"><button>Retour</button></a>

</body>
</html>

File: modifier_professeur.py
import sys
from PyQt5.QtWidgets import QApplication, QWidget, QLabel, QLineEdit, QPushButton, QVBoxLayout, QFormLayout, QDialog, QDialogButtonBox, QComboBox, QHBoxLayout, QMessageBox
from PyQt5.QtCore import Qt
import mysql.connector

class ModifierProfesseurDialog(QDialog):
    def __init__(self, professeur_id):
        super().__init__()

        self.professeur_id = professeur_id

        self.setWindowTitle("Modifier un professeur")
        self.layout = QVBoxLayout()

        self.initUI()

        self.setLayout(self.layout)

    def initUI(self):
        self.form_layout = QFormLayout()

        self.nom_input = QLineEdit()
        self.form_layout.addRow("Nom:", self.nom_input)

        self.prenom_input = QLineEdit()
        self.form_layout.addRow("Prénom:", self.prenom_input)

        self.situation_matrimoniale_input = QComboBox()
        self.situation_matrimoniale_input.addItems(["Célibataire", "Marié(e)", "Veuf(ve)"])
        self.form_layout.addRow("Situation Matrimoniale:", self.situation_matrimoniale_input)

        self.lieu_naissance_input = QLineEdit()
        self.form_layout.addRow("Lieu de Naissance:", self.lieu_naissance_input)

        self.date_naissance_input = QLineEdit()
        self.form_layout.addRow("Date de Naissance:", self.date_naissance_input)

        self.diplome_formation_input = QLineEdit()
        self.form_layout.addRow("Diplôme de Formation:", self.diplome_formation_input)

        self.autre_diplome_input = QLineEdit()
        self.form_layout.addRow("Autre Diplôme:", self.autre_diplome_input)

        self.matieres_enseignees_input = QLineEdit()
        self.form_layout.addRow("Matières Enseignées:", self.matieres_enseignees_input)

        self.autre_matiere_input = QLineEdit()
        self.form_layout.addRow("Autre Matière:", self.autre_matiere_input)

        self.login_input = QLineEdit()
        self.form_layout.addRow("Login:", self.login_input)

        self.passwd_input = QLineEdit()
        self.passwd_input.setEchoMode(QLineEdit.Password)
        self.form_layout.addRow("Mot de passe:", self.passwd_input)

        self.submit_button = QPushButton("Enregistrer")
        self.submit_button.clicked.connect(self.update_professeur)
        self.layout.addLayout(self.form_layout)
        self.layout.addWidget(self.submit_button)

    def update_professeur(self):
        nom = self.nom_input.text()
        prenom = self.prenom_input.text()
        situation_matrimoniale = self.situation_matrimoniale_input.currentText()
        lieu_naissance = self.lieu_naissance_input.text()
        date_naissance = self.date_naissance_input.text()
        diplome_formation = self.diplome_formation_input.text()
        autre_diplome = self.autre_diplome_input.text()
        matieres_enseignees = self.matieres_enseignees_input.text()
        autre_matiere = self.autre_matiere_input.text()
        login = self.login_input.text()
        passwd = self.passwd_input.text()

        try:
            db = mysql.connector.connect(
                host="localhost",
                user="root",
                password="",
                database="gdn"
            )
            cursor = db.cursor()

            query = "UPDATE professeurs SET Nom_Professeur = %s, Prenom_Professeur = %s, Situation_Matrimoniale = %s, Lieu_Naissance = %s, Date_Naissance = %s, Diplome_Formation = %s, Autre_Diplome = %s, Matieres_Enseignees = %s, Autre_Matiere = %s, Login_Professeur = %s, Passwd_Professeur = %s WHERE ID_Professeur = %s"
            data = (nom, prenom, situation_matrimoniale, lieu_naissance, date_naissance, diplome_formation, autre_diplome, matieres_enseignees, autre_matiere, login, passwd, self.professeur_id)
            cursor.execute(query, data)
            db.commit()
            db.close()

            QMessageBox.information(self, "Success", "Les informations du professeur ont été mises à jour avec succès.")
            self.accept()
        except mysql.connector.Error as err:
            QMessageBox.critical(self, "Error", f"Erreur : {err}")

class MainWindow(QWidget):
    def __init__(self):
        super().__init__()

        self.setWindowTitle("Modifier un professeur")
        self.layout = QVBoxLayout()

        self.initUI()

        self.setLayout(self.layout)

    def initUI(self):
        self.id_label = QLabel("ID du professeur:")
        self.id_input = QLineEdit()
        self.id_input.setPlaceholderText("Entrez l'ID du professeur")

        self.load_button = QPushButton("Charger")
        self.load_button.clicked.connect(self.load_professeur)

        self.layout.addWidget(self.id_label)
        self.layout.addWidget(self.id_input)
        self.layout.addWidget(self.load_button)

    def load_professeur(self):
        professeur_id = self.id_input.text()
        if professeur_id:
            dialog = ModifierProfesseurDialog(professeur_id)
            dialog.exec_()
        else:
            QMessageBox.warning(self, "Warning", "Veuillez entrer l'ID du professeur.")

if __name__ == "__main__":
    app = QApplication(sys.argv)
    window = MainWindow()
    window.show()
    sys.exit(app.exec_())
